src/ln/router.rs

```diff
@@ -68,6 +70,9 @@ struct ChannelInfo {
 	features: GlobalFeatures,
 	one_to_two: DirectionalChannelInfo,
 	two_to_one: DirectionalChannelInfo,
+	//this is cached here so we can send out it later if required by route_init_sync
+	announcement_message : Option<msgs::ChannelAnnouncement>,
```

Comment:
This is very redundant - most of the fields are stored in features or the map that points to the ChannelInfo - we should only be storing the non-redundant bits so we can reconstruct the ChannelAnnouncement message (and should not be storing that info for ChannelAnnouncements that have extra bytes on the end we didn't parse - we simply should not forward such messages).

Reply:
The reason I did this is to try and simplify the implementation, by storing the complete field as one, we dont have to reconstruct it/store more fields, the fields we have to add is the largest in the packets, for example:
channel_announcement:
total size : 463 bytes (not counting address len).
We currently store : 107 bytes.
So we only waste 107 bytes per message. 

This implementation I believe is cleaner in code, but I can only add the fields we additional need?

Reply:
No it isnt, the cleaner code I am after, those 4 fields are by far the largest fields. You waste 107 byts in the example but you save 16 lines of code (11 on building and 5 on storing). Thats the part I am more interested in.

Reply:
I mean its obviously not a huge deal either way, personally I'd prefer to not waste an extra 107 bytes per entry here as there may eventually be a lot of entries, but if you feel strongly feel free to leave it.

Reply:
I am going to add a comment there to keep an eye on those. If its a problem we should go for the more efficient route.